PhysicsTools/PatAlgos/python/producersHeavyIons/heavyIonJets_cff.py

```diff
@@ -1,19 +1,71 @@
 import FWCore.ParameterSet.Config as cms
 
-from PhysicsTools.PatAlgos.producersLayer1.jetProducer_cff import *
-from RecoHI.HiJetAlgos.HiGenCleaner_cff import *
-heavyIonCleaned = cms.Sequence(genPartons*hiPartons+heavyIonCleanedGenJets)
+from RecoJets.Configuration.GenJetParticles_cff import genParticlesForJets
+from RecoHI.HiJetAlgos.HiGenCleaner_cff import hiPartons
+from RecoHI.HiJetAlgos.HiGenJets_cff import ak4HiGenJets
+from RecoHI.HiJetAlgos.HiGenCleaner_cff import heavyIonCleanedGenJets
 
-makeHeavyIonJets = cms.Sequence(
-    patJetCorrections *
-
-    # pat and HI specifics
-    heavyIonCleaned *
-    patJetGenJetMatch *
-    patJetPartonMatch *
+allPartons = cms.EDProducer(
+    "PartonSelector",
+    src = cms.InputTag('genParticles'),
+    withLeptons = cms.bool(False),
+    )
 
-    # object production
-    patJets
+cleanedPartons = hiPartons.clone(
+    src = 'allPartons',
     )
 
+ak4HiCleanedGenJets = heavyIonCleanedGenJets.clone(src = "ak4HiGenJets")
+
+cleanedGenJetsTask = cms.Task(
+    genParticlesForJets,
+    cleanedPartons,
+    ak4HiGenJets,
+    ak4HiCleanedGenJets
+)
+
+from RecoHI.HiJetAlgos.HiRecoPFJets_cff import PFTowers, pfEmptyCollection, ak4PFJetsForFlow, hiPuRho, hiFJRhoFlowModulation, akCs4PFJets
+from RecoHI.HiTracking.highPurityGeneralTracks_cfi import highPurityGeneralTracks
+from RecoJets.JetAssociationProducers.ak5JTA_cff import *
+from RecoBTag.Configuration.RecoBTag_cff import impactParameterTagInfos, trackCountingHighEffBJetTags, trackCountingHighPurBJetTags, jetProbabilityBJetTags, jetBProbabilityBJetTags, secondaryVertexTagInfos, combinedSecondaryVertexV2BJetTags, simpleSecondaryVertexHighEffBJetTags, simpleSecondaryVertexHighPurBJetTags
+ak5JetTracksAssociatorAtVertex
+from RecoBTag.SecondaryVertex.simpleSecondaryVertex2TrkComputer_cfi import *
+from RecoBTag.SecondaryVertex.simpleSecondaryVertex3TrkComputer_cfi import *
+from RecoBTag.SecondaryVertex.combinedSecondaryVertexV2Computer_cfi import *
+from RecoBTag.ImpactParameter.jetBProbabilityComputer_cfi import *
+from RecoBTag.ImpactParameter.jetProbabilityComputer_cfi import *
+from RecoBTag.ImpactParameter.trackCounting3D2ndComputer_cfi import *
+from RecoBTag.ImpactParameter.trackCounting3D3rdComputer_cfi import *
+from PhysicsTools.PatAlgos.recoLayer0.jetCorrFactors_cfi import *
 
+ak5JetTracksAssociatorAtVertex.jets = "akCs4PFJets"
+ak5JetTracksAssociatorAtVertex.tracks = "highPurityGeneralTracks"
+
+impactParameterTagInfos.jetTracks = "ak5JetTracksAssociatorAtVertex"
```

Comment:
global replace to existing modules is rather dangerous.
Please modify the module parameters in place of their definitions using the HI modifiers

Reply:
Just to be clear, keep the setting of these parameters in the current location, but just use the modifiers, right?

Reply:
I meant to say to modify where the modules are defined (similar pattern as in other files in this PR); so, not in this file.